vision_agent_tools/tools/shared_model_manager.py

```diff
@@ -60,37 +58,28 @@ async def fetch_model(self, model_id: str) -> BaseTool:
             raise ValueError(f"Model '{model_id}' not found in the pool.")
 
         model = self.models[model_id]
-        lock = self.model_locks[model_id]
-        device = self.devices[model_id]
-
-        async def get_model_with_lock() -> Any:
-            async with lock:
-                if device == Device.GPU:
-                    # Acquire semaphore if needed
-                    async with self.gpu_semaphore:
-                        # Move existing GPU model to CPU
-                        exisitng = self._get_current_gpu_model()
-                        if exisitng:
-                            await self._move_to_cpu(exisitng)
-
-                        # Update current GPU model
-                        self.current_gpu_model = model_id
-                        model.to(Device.GPU)
-                return model
-
-        return await get_model_with_lock()
+
+        # Move existing GPU model to CPU
+        exisitng = self._get_current_gpu_model()
+        if exisitng:
+            self._move_to_cpu(exisitng)
+
+        # Update current GPU model
+        self.current_gpu_model = model_id
+        model.to(Device.GPU)
+        return model
 
     def _get_current_gpu_model(self) -> Optional[str]:
         """
         Returns the class name of the model currently using the GPU (if any).
         """
         return self.current_gpu_model
 
-    async def _move_to_cpu(self, class_name: str) -> None:
+    def _move_to_cpu(self, model_id: str) -> None:
         """
         Moves a model to CPU and releases the GPU semaphore (if held).
         """
-        model = self.models[class_name]
+        model = self.models[model_id]
         model.to(Device.CPU)
-        if self.current_gpu_model == class_name:
+        if self.current_gpu_model == model_id:
             self.current_gpu_model = None
```

Comment:
I suggest to do this assignment (self.current_gpu_model) all in `fetch_model` since you are manipulating this var there too